rueidislimiter/limiter.go

```diff
@@ -0,0 +1,243 @@
+package rueidislimiter
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/redis/rueidis"
+)
+
+type Result struct {
+	// Allowed indicates if the request is allowed under the rate limit
+	Allowed bool
+	// Remaining is the number of remaining tokens in the current window
+	Remaining int64
+	// ResetAfter is the duration until the rate limit resets
+	ResetAfter time.Duration
+	// RetryAfter is the duration after which the request may be retried (0 if allowed)
+	RetryAfter time.Duration
+}
+
+type RateLimiterClient interface {
+	Check(ctx context.Context, identifier string) (Result, error)
+	Allow(ctx context.Context, identifier string) (Result, error)
+	AllowN(ctx context.Context, identifier string, n int64) (Result, error)
+}
+
+const PlaceholderPrefix = "rueidislimiter"
+
+type rateLimiter struct {
+	client rueidis.Client
+	key    string
+	limit  int
+	window time.Duration
+}
+
+type RateLimiterOption struct {
```

Comment:
I think, we need expiration duration option need. 

some cases, for example video contents, become fashionable and not long after can be out of fashion. In this case, dead rate limiter still exists wasting memory.

Only growing, never shrinking data set size is not a good idea. how about adding TTL option which must be bigger than window?

Reply:
Oh, i missed the code!